lib/administrate/field/base.rb

```diff
@@ -52,6 +52,22 @@ def name
         attribute.to_s
       end
 
+      def read_value(data = nil)
+        if options.key?(:getter)
+          if options[:getter].respond_to?(:call)
+            options[:getter].call(self)
+          else
+            resource&.public_send(options[:getter])
+          end
+        else
+          if data.nil?
+            resource&.public_send(attribute)
+          else
+            data
+          end
+        end
```

Comment:
Unfortunately, the example I mentioned with `Foobar` doesn't work: `public_send` fails because the method doesn't exist in the resource. Note that the safe navigation operator (`&.`) doesn't help here because it's useful only when the resource is nil.

However, I think it can be fixed with a `respond_to?` instead, as below:

```suggestion
        elsif data.nil?
          resource.respond_to?(attribute) ? resource.public_send(attribute) : nil
        else
          data
        end
      end
```

This should also fix the linter error asking for `elsif` to be used.

Reply:
Apologies for the oversight. I’ve made the fixes and checked with standardrb.
I reviewed the your reference code and felt that `try` would be sufficient, so I rewrote it using `try`. Do you have any concerns about this approach?